packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -620,16 +623,20 @@ export default class NewVm extends BaseComponent {
         )
     }
   )
-  _getDefaultNetworkId = () => {
-    const resourceSet = this._getResolvedResourceSet()
-    if (resourceSet) {
-      const { network } = resourceSet.objectsByType
-      return !isEmpty(network) && network[0].id
+  _getDefaultNetworkId = template => {
+    if (template === undefined) {
+      return
     }
-    const network = find(this._getPoolNetworks(), network => {
-      const pif = getObject(store.getState(), network.PIFs[0])
-      return pif && pif.management
-    })
+
+    const resourceSet = this._getResolvedResourceSet()
+    const network = resourceSet
```

Comment:
Let's check against `this.state.state.pool` to determine if we're in Admin or Self Service mode, as this is what we usually did so far.